### ISSUE TITLE
Can not expose function under namespace that contians '.' (global.Math for example) on C-API

### ISSUE THREAD

**Issue by MeirShpilraien** [wasmerio/wasmer#1061]
### Summary
Not sure if its a bug or maybe I am doing something wrong. Here is my code and for some reason I am getting this error: `(1 of 57) Import not found, namespace: global.Math, name: pow` though I am pretty sure that I exposed the global.Math pow function.
```
// Create a new func to hold the parameter and signature
    // of our `print_str` host function
    wasmer_value_tag params_sig[] = {WASM_I32, WASM_I32};
    wasmer_value_tag returns_sig[] = {WASM_I32};
    wasmer_import_func_t *func = wasmer_import_func_new((void (*)(void *)) my_pow, params_sig, 2, returns_sig, 1);

    const char *module_name = "global.Math";
    wasmer_byte_array module_name_bytes = { .bytes = (const uint8_t *) module_name,
                                            .bytes_len = strlen(module_name) };

    // Define a function import
    const char *import_name = "pow";
    wasmer_byte_array import_name_bytes = { .bytes = (const uint8_t *) import_name,
                                            .bytes_len = strlen(import_name) };
    wasmer_import_t func_import = { .module_name = module_name_bytes,
                                    .import_name = import_name_bytes,
                                    .tag = WASM_FUNCTION,
                                    .value.func = func };

    // Define a memory import
    const char *import_memory_name = "memory";
    wasmer_byte_array import_memory_name_bytes = { .bytes = (const uint8_t *) import_memory_name,
                                                   .bytes_len = strlen(import_memory_name) };
    wasmer_import_t memory_import = { .module_name = module_name_bytes,
                                      .import_name = import_memory_name_bytes,
                                      .tag = WASM_MEMORY };
    wasmer_memory_t *memory = NULL;
    wasmer_limit_option_t max = { .has_some = true,
                                  .some = 256 };
    wasmer_limits_t descriptor = { .min = 256,
                                   .max = max };

    wasmer_result_t memory_result = wasmer_memory_new(&memory, descriptor);
    if (memory_result != WASMER_OK)
    {
        int error_len = wasmer_last_error_length();
        char *error_str = RG_ALLOC(error_len);
        wasmer_last_error_message(error_str, error_len);
        RedisModule_ReplyWithError(ctx, error_str);
        RG_FREE(error_str);
        return REDISMODULE_OK;
    }
    memory_import.value.memory = memory;

    // Define a global import
    const char *import_global_name = "__memory_base";
    wasmer_byte_array import_global_name_bytes = { .bytes = (const uint8_t *) import_global_name,
                                                   .bytes_len = strlen(import_global_name) };
    wasmer_import_t global_import = { .module_name = module_name_bytes,
                                      .import_name = import_global_name_bytes,
                                      .tag = WASM_GLOBAL };

    wasmer_value_t val = { .tag = WASM_I32,
                           .value.I32 = 1024 };
    wasmer_global_t *global = wasmer_global_new(val, false);
    global_import.value.global = global;

    wasmer_import_t imports[] = {func_import, global_import, memory_import};

  size_t len;
    const char* bytes = RedisModule_StringPtrLen(argv[1], &len);
    wasmer_module_t *module1;
    wasmer_result_t compile_result = wasmer_compile(&module1, (char*)bytes, len);


    if (compile_result != WASMER_OK)
    {
        int error_len = wasmer_last_error_length();
        char *error_str = RG_ALLOC(error_len);
        wasmer_last_error_message(error_str, error_len);
        RedisModule_ReplyWithError(ctx, error_str);
        RG_FREE(error_str);
        return REDISMODULE_OK;
    }


    wasmer_instance_t *instance = NULL;
    wasmer_result_t initiate_result = wasmer_module_instantiate(module1, &instance, imports, 3);

    if (initiate_result != WASMER_OK)
    {
        int error_len = wasmer_last_error_length();
        char *error_str = RG_ALLOC(error_len);
        wasmer_last_error_message(error_str, error_len);
        RedisModule_ReplyWithError(ctx, error_str);
        RG_FREE(error_str);
        return REDISMODULE_OK;
    }
```

If I am trying to expose any other missing function that does not contain `.` in its namespace then its working just fine. 

### Additional details

I need to expose those functions because I can not use the Emscripten implementation from the C-API (https://github.com/wasmerio/wasmer/issues/574)

**Comment by syrusakbary** [comment]
Thanks for the report, we will take a look soon!

**Comment by MarkMcCaskey** [comment]
Okay I just did an investigation of this!

I think there are a few small issues here on your side:

1. You're using `module_name_bytes` for all imports, so it can't find things like `memory` because that should be in `"env"`, not in `"global.Math"`
2. Emscripten expects imports to be prefixed with a `_`, so `const char *import_name = "pow";` should become `const char *import_name = "_pow";`.

I got it working on my side from your example by both including `math.h` and using their pow as well as declaring my own extern pow function and editing the Wasm to force it into the `global.Math` namespace.

Feel free to respond or reopen this if that doesn't work for you!
